src/main/java/io/jawg/osmcontributor/ui/activities/PhotoActivity.java

```diff
@@ -352,17 +358,18 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
     @Subscribe(threadMode = ThreadMode.MAIN)
     public void onPhotosFoundEvent(PhotosFoundEvent photosFoundEvent) {
         List<List<Size>> photos = photosFoundEvent.getPhotos();
-        if (photos != null && !photos.isEmpty()) {
+        if (photos != null && !photos.isEmpty() && photosFoundEvent.getPoiId().equals(poiId)) {
```

Comment:
No onPhotosFoundEvent should be thrown if the user has changed POI.
==> There should be a way to cancel the loading of images if the user switches views.
If this is too tricky, forget about it.